src/Microsoft.DotNet.Tools.Builder/CompileContext.cs

```diff
@@ -406,6 +406,7 @@ public CompilerIO(List<string> inputs, List<string> outputs)
             var compilerIO = new CompilerIO(new List<string>(), new List<string>());
             var calculator = project.GetOutputPathCalculator(outputPath);
             var binariesOutputPath = calculator.GetOutputDirectoryPath(buildConfiguration);
+            var intermediateOutputPath = calculator.GetIntermediateOutputDirectoryPath(buildConfiguration, intermediaryOutputPath);
```

Comment:
Should we change these variable names? intermediate and intermediary and pretty hard to distinguish. Maybe use just one and re-assign here?